src/odemis/driver/piezomotor.py

```diff
@@ -1138,7 +1138,8 @@ def _parseMessage(self, msg):
 
     def move(self, steps, target_pos):
         # simple move, same duration for every length, don't care about speed
-        self.executor.submit(self._do_move, steps, target_pos)
+        f = self.executor.submit(self._do_move, steps, target_pos)
+        f.result(timeout=900)
```

Comment:
I don't think that's correct. Because now the command is blocking. Typically the hardware accepts the command, and immediately returns.

If the driver needs to wait until the move is complete. It should be done in the actual driver class, in `_waitEndMotion()`. Or maybe this part is not correctly simulated?